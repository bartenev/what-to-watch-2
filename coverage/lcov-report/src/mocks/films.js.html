
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/mocks/films.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">src/mocks</a> films.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>45/45</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>6/6</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>42/42</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a>
<a name='L107'></a><a href='#L107'>107</a>
<a name='L108'></a><a href='#L108'>108</a>
<a name='L109'></a><a href='#L109'>109</a>
<a name='L110'></a><a href='#L110'>110</a>
<a name='L111'></a><a href='#L111'>111</a>
<a name='L112'></a><a href='#L112'>112</a>
<a name='L113'></a><a href='#L113'>113</a>
<a name='L114'></a><a href='#L114'>114</a>
<a name='L115'></a><a href='#L115'>115</a>
<a name='L116'></a><a href='#L116'>116</a>
<a name='L117'></a><a href='#L117'>117</a>
<a name='L118'></a><a href='#L118'>118</a>
<a name='L119'></a><a href='#L119'>119</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">954x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">75x</span>
<span class="cline-any cline-yes">75x</span>
<span class="cline-any cline-yes">192x</span>
<span class="cline-any cline-yes">192x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">75x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">57x</span>
<span class="cline-any cline-yes">57x</span>
<span class="cline-any cline-yes">57x</span>
<span class="cline-any cline-yes">57x</span>
<span class="cline-any cline-yes">57x</span>
<span class="cline-any cline-yes">57x</span>
<span class="cline-any cline-yes">57x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">57x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">102x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">36x</span>
<span class="cline-any cline-yes">36x</span>
<span class="cline-any cline-yes">36x</span>
<span class="cline-any cline-yes">123x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">36x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {Genres} from "../const";
&nbsp;
const getRandomNumber = (max, min = 0) =&gt; {
  return min + Math.round(Math.random() * (max - min));
};
&nbsp;
const names = [`Vasya`, `Kolya`, `Ivan`, `Petr`, `Genka`, `Cheba`, `Ilya`, `Karina`];
&nbsp;
const directors = [
  `Christopher Nolan`,
  `Steven Spielberg`,
  `Quentin Tarantino`,
  `Martin Scorsese`,
  `David Fincher`,
  `Ridley Scott`,
  `Stanley Kubrick`,
  `Robert Zemeckis`,
  `Francis Ford Coppola`,
  `Clint Eastwood`
];
&nbsp;
const genres = Object.values(Genres).filter((genre) =&gt; genre !== Genres.ALL_GENRES);
&nbsp;
const text = `
Lorem ipsum dolor sit amet, consectetur adipiscing elit.
Praesent in nisi condimentum, aliquet nisl at, fermentum neque.
Vivamus ex urna, tincidunt eget feugiat a, luctus ut arcu.
Vivamus elementum nisl eu elit cursus, id luctus justo posuere.
Integer commodo enim eget ullamcorper faucibus. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
Mauris interdum dolor a convallis pharetra. Fusce porttitor dictum mi, id tincidunt ligula lobortis eu.
Nunc sapien turpis, eleifend in odio at, ullamcorper efficitur lorem.
Quisque fringilla lacinia pulvinar. Donec vel semper nunc, ac posuere lorem. Nulla sed tortor nunc.`;
&nbsp;
const textItems = text.split(`.\n`);
&nbsp;
const getRandomText = () =&gt; {
  let randomText = ``;
  for (let i = 0; i &lt; getRandomNumber(5, 1); i++) {
    randomText += textItems[getRandomNumber(textItems.length - 1)];
    randomText += `. `;
  }
  return randomText;
};
&nbsp;
const getRandomDate = () =&gt; {
  const targetDate = new Date();
  const rangeDays = 100;
  const diffValue = -1 * getRandomNumber(rangeDays, 0);
  targetDate.setDate(targetDate.getDate() + diffValue);
  targetDate.setHours(getRandomNumber(23, 0));
  targetDate.setMinutes(getRandomNumber(59, 0));
  return targetDate;
};
&nbsp;
const getReview = () =&gt; {
  return {
    name: names[getRandomNumber(names.length - 1)],
    date: getRandomDate(),
    text: getRandomText(),
    rating: getRandomNumber(10),
  };
};
&nbsp;
const getRandomElementOfArray = (array) =&gt; {
  return array[getRandomNumber(array.length - 1)];
};
&nbsp;
const getRandomArrayOfSomething = (getElement, max, min, array = []) =&gt; {
  const count = getRandomNumber(max, min);
  let randomArray = [];
  for (let i = 0; i &lt; count; i++) {
    randomArray.push(getElement(array));
  }
&nbsp;
  return randomArray;
};
&nbsp;
const filmsNames = [`Bohemian Rhapsody`, `Macbeth`, `Aviator`, `We need to talk about Kevin`, `What We Do in the Shadows`,
  `Revenant`, `Johnny English`, `Shutter Island`, `Pulp Fiction`, `No Country for Old Men`, `Snatch`, `Moonrise Kingdom`,
  `Seven Years in Tibet`, `War of the Worlds`, `Dardjeeling Limited`, `Orlando`, `Mindhunter`, `Midnight Special`];
&nbsp;
const getFilms = () =&gt; {
  return filmsNames.map((filmName) =&gt; {
    const score = getRandomNumber(10, 0);
    let level = `Excellent`;
&nbsp;
    if (score &lt;= 6) {
      level = `Bad`;
    } else if (score &lt;= 8) {
      level = `Good`;
    }
&nbsp;
    return ({
      title: filmName,
      src: {
        poster: `img/${filmName.toLowerCase().replace(/ /g, `-`)}.jpg`,
        preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        video: ``,
      },
      director: getRandomElementOfArray(directors),
      starring: getRandomArrayOfSomething(getRandomElementOfArray, 6, 1, directors),
      reviews: getRandomArrayOfSomething(getReview, 6, 0),
      runTime: getRandomNumber(300, 1),
      genre: getRandomElementOfArray(genres),
      released: getRandomNumber(2021, 1950),
      description: getRandomText(),
      rating: {
        score,
        level,
        count: getRandomNumber(10000),
      },
    });
  });
};
&nbsp;
const films = getFilms();
&nbsp;
export default films;
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Sun Oct 17 2021 13:06:19 GMT+0300 (Moscow Standard Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    