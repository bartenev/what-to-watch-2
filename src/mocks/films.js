import {Genres} from "../const";

const getRandomNumber = (max, min = 0) => {
  return min + Math.round(Math.random() * (max - min));
};

const names = [`Vasya`, `Kolya`, `Ivan`, `Petr`, `Genka`, `Cheba`, `Ilya`, `Karina`];

const directors = [
  `Christopher Nolan`,
  `Steven Spielberg`,
  `Quentin Tarantino`,
  `Martin Scorsese`,
  `David Fincher`,
  `Ridley Scott`,
  `Stanley Kubrick`,
  `Robert Zemeckis`,
  `Francis Ford Coppola`,
  `Clint Eastwood`
];

const genres = Object.values(Genres);

const text = `
Lorem ipsum dolor sit amet, consectetur adipiscing elit.
Praesent in nisi condimentum, aliquet nisl at, fermentum neque.
Vivamus ex urna, tincidunt eget feugiat a, luctus ut arcu.
Vivamus elementum nisl eu elit cursus, id luctus justo posuere.
Integer commodo enim eget ullamcorper faucibus. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
Mauris interdum dolor a convallis pharetra. Fusce porttitor dictum mi, id tincidunt ligula lobortis eu.
Nunc sapien turpis, eleifend in odio at, ullamcorper efficitur lorem.
Quisque fringilla lacinia pulvinar. Donec vel semper nunc, ac posuere lorem. Nulla sed tortor nunc.`;

const textItems = text.split(`.\n`);

const getRandomText = () => {
  let randomText = ``;
  for (let i = 0; i < getRandomNumber(5, 1); i++) {
    randomText += textItems[getRandomNumber(textItems.length - 1)];
    randomText += `. `;
  }
  return randomText;
};

const getRandomDate = () => {
  const targetDate = new Date();
  const rangeDays = 100;
  const diffValue = -1 * getRandomNumber(rangeDays, 0);
  targetDate.setDate(targetDate.getDate() + diffValue);
  targetDate.setHours(getRandomNumber(23, 0));
  targetDate.setMinutes(getRandomNumber(59, 0));
  return targetDate;
};

const getReview = () => {
  return {
    name: names[getRandomNumber(names.length - 1)],
    date: getRandomDate(),
    text: getRandomText(),
    rating: getRandomNumber(10),
  };
};

const getRandomElementOfArray = (array) => {
  return array[getRandomNumber(array.length - 1)];
};

const getRandomArrayOfSomething = (getElement, max, min, array = []) => {
  const count = getRandomNumber(max, min);
  let randomArray = [];
  for (let i = 0; i < count; i++) {
    randomArray.push(getElement(array));
  }

  return randomArray;
};

const filmsNames = [`Bohemian Rhapsody`, `Macbeth`, `Aviator`, `We need to talk about Kevin`, `What We Do in the Shadows`,
  `Revenant`, `Johnny English`, `Shutter Island`, `Pulp Fiction`, `No Country for Old Men`, `Snatch`, `Moonrise Kingdom`,
  `Seven Years in Tibet`, `Midnight Special`, `War of the Worlds`, `Dardjeeling Limited`, `Orlando`, `Mindhunter`, `Midnight Special`];

const getFilms = () => {
  return filmsNames.map((filmName) => {
    const score = getRandomNumber(10, 0);
    let level = `Excellent`;

    if (score <= 6) {
      level = `Bad`;
    } else if (score <= 8) {
      level = `Good`;
    }

    return ({
      title: filmName,
      src: {
        poster: `img/${filmName.toLowerCase().replace(/ /g, `-`)}.jpg`,
        preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        video: ``,
      },
      director: getRandomElementOfArray(directors),
      starring: getRandomArrayOfSomething(getRandomElementOfArray, 6, 1, directors),
      reviews: getRandomArrayOfSomething(getReview, 6, 0),
      runTime: getRandomNumber(300, 1),
      genre: getRandomElementOfArray(genres),
      released: getRandomNumber(2021, 1950),
      description: getRandomText(),
      rating: {
        score,
        level,
        count: getRandomNumber(10000),
      },
    });
  });
};

const films = getFilms();

export default films;
