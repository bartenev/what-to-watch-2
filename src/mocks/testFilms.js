export const testFilms =
  [
    {
      id: 3,
      title: `Pulp Fiction`,
      src: {
        poster: `img/pulp-fiction.jpg`,
        backgroundImage: `img/pulp-fiction.jpg`,
        previewImage: `img/pulp-fiction.jpg`,
        previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      },
      director: `Christopher Nolan`,
      starring: [
        `Quentin Tarantino`,
        `David Fincher`,
      ],
      runTime: 240,
      genre: `Comedies`,
      released: 1956,
      description: `Mauris interdum dolor a convallis pharetra. Fusce porttitor dictum mi, id tincidunt ligula lobortis eu. Integer commodo enim eget ullamcorper faucibus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer commodo enim eget ullamcorper faucibus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. `,
      rating: {
        score: 5,
        count: 783,
      },
      isFavorite: true,
      backgroundColor: `#123456`,
    },
    {
      id: 7,
      title: `Macbeth`,
      src: {
        poster: `img/macbeth.jpg`,
        backgroundImage: `img/macbeth.jpg`,
        previewImage: `img/macbeth.jpg`,
        previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      },
      director: `Christopher Nolan`,
      starring: [
        `Robert Zemeckis`,
        `Christopher Nolan`,
        `Clint Eastwood`,
      ],
      runTime: 220,
      genre: `Crime`,
      released: 1966,
      description: `Nunc sapien turpis, eleifend in odio at, ullamcorper efficitur lorem. Vivamus ex urna, tincidunt eget feugiat a, luctus ut arcu. Mauris interdum dolor a convallis pharetra. Fusce porttitor dictum mi, id tincidunt ligula lobortis eu. `,
      rating: {
        score: 4,
        count: 6354,
      },
      isFavorite: false,
      backgroundColor: `#987654`,
    },
  ];
