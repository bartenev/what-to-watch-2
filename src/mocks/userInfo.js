import {AuthorizationStatus} from "../reducer/user/user";

export const UserInfo = {
  authorizationStatus: AuthorizationStatus.AUTH,
  userInfo: {
    id: 5,
    email: `nikita@gmail.com`,
    name: `nikita`,
    avatar: `url`,
  }
};
