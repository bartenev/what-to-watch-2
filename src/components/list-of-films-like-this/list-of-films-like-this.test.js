import React from "react";
import renderer from "react-test-renderer";
import ListOfFilmsLikeThis from "./list-of-films-like-this";
import {testFilms} from "../../mocks/testFilms";

it(`ListOfFilmsLikeThis is rendered correctly`, () => {
  const tree = renderer.create((
    <ListOfFilmsLikeThis
      currentFilm={testFilms[0]}
      films={testFilms}
      onHover={() => {}}
      onClick={() => {}}
    />
  ), {
    createNodeMock: () => {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
