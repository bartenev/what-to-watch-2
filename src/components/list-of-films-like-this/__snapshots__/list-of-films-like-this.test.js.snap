// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfFilmsLikeThis is rendered correctly 1`] = `null`;
